vscode-ext/src/StatusBarComponent.ts

```diff
@@ -38,36 +47,138 @@ export class StatusBarComponent {
+            this.verifyStatus.insert(document, results);
             this.render();
         });
+
     }
 
     render() {
         if (this.enabled) {
-            switch (this.status) {
+
+            switch (this.serverStatus) {
+                case ServerStatus.NotStarted:
+                    this.view.text = "$(debug-start) Caesar Inactive";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipBase = stoppedTooltipBase;
+                    break;
                 case ServerStatus.Stopped:
-                    this.view.text = "$(debug-stop) Et tu, Brute?";
+                    this.view.text = "$(debug-start) Et tu, Brute?";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipBase = stoppedTooltipBase;
                     break;
                 case ServerStatus.FailedToStart:
                     this.view.text = "$(warning) Failed to start Caesar";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipBase = stoppedTooltipBase;
                     break;
                 case ServerStatus.Starting:
-                    this.view.text = "$(sync~spin) Starting Caesar...";
+                    this.view.text = "$(loading~spin) Starting Caesar...";
+                    this.view.command = ""
                     break;
                 case ServerStatus.Ready:
-                    this.view.text = "$(check) Caesar Ready";
+                    this.view.text = "$(thumbsup-filled) Caesar Ready";
+                    this.view.command = "caesar.showOutput";
+                    this.tooltipBase = runningTooltipBase;
                     break;
                 case ServerStatus.Verifying:
                     this.view.text = "$(sync~spin) Verifying...";
+                    this.view.command = ""
                     break;
                 case ServerStatus.Finished:
-                    this.view.text = "$(check) Verified";
+                    this.tooltipBase = runningTooltipBase;
+                    this.handleFinishedStatus();
+                    this.view.command = "caesar.showOutput";
                     break;
             }
+
+            this.renderTooltip();
             this.view.show();
         } else {
             this.view.hide();
         }
     }
+
+    private extendedTooltip(extension: vscode.MarkdownString) {
+        return new vscode.MarkdownString(this.tooltipBase.value + "\n\n --- \n" + extension.value, true);
+    }
+
+    private renderTooltip() {
+        if (this.tooltipExtension.value === "") {
+            this.view.tooltip = this.tooltipBase;
+            this.view.tooltip.isTrusted = true;
+            return;
+        }
+        this.view.tooltip = this.extendedTooltip(this.tooltipExtension);
+        this.view.tooltip.isTrusted = true;
+    }
+
+    private countResults(results: [Range, VerifyResult][]): [number, number, number] {
```

Comment:
I don't see a reason why this is a member function. Either make it a freestanding function or just inline its definition as well (I prefer the latter).